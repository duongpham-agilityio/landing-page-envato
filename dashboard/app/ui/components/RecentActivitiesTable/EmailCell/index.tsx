// Libs
import { memo } from 'react';
import { Td, Text } from '@chakra-ui/react';
import Link from 'next/link';

interface EmailCellProps {
  email: string;
}

const EmailCell = ({ email }: EmailCellProps): JSX.Element => (
  <Td
    py={5}
    pr={5}
    pl={0}
    fontSize="md"
    color="text.primary"
    fontWeight="semibold"
    textAlign="left"
    w={{ base: 150, md: 20 }}
  >
    <Text
      as={Link}
      href={`mailto:${email}`}
      fontSize="md"
      fontWeight="semibold"
      whiteSpace="break-spaces"
      noOfLines={1}
      w={{ base: 100, md: 220, '3xl': 200, '5xl': 200, '7xl': 350 }}
      flex={1}
    >
      {email}
    </Text>
  </Td>
);

const EmailCellMemorized = memo(EmailCell);

export default EmailCellMemorized;
