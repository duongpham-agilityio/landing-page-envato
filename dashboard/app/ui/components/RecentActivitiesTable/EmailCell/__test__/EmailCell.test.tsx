import { render } from '@testing-library/react';
import { Table } from '@chakra-ui/react';

// Components
import EmailCell from '..';

const mockProps = {
  email: 'user01@gmail.com',
};

describe('EmailCell component', () => {
  const setup = () =>
    render(<EmailCell {...mockProps} />, {
      wrapper: Table,
    });

  it('should match with snapshot', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
});
