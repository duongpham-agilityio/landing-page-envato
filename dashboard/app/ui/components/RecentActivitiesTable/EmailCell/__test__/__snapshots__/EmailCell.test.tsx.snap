// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailCell component should match with snapshot 1`] = `
<div>
  <table
    class="chakra-table css-0"
  >
    <td
      class="css-oe3t8o"
    >
      <a
        class="chakra-text css-f498oj"
        href="mailto:user01@gmail.com"
      >
        user01@gmail.com
      </a>
    </td>
  </table>
</div>
`;
